nncf/torch/pruning/filter_pruning/global_ranking/evolutionary_optimization.py

```diff
@@ -270,7 +270,7 @@ class LeGRPruner:
     and resetting all changes in the model made by the environment.
     """
 
-    def __init__(self, filter_pruner_ctrl: "FilterPruningController", model: nn.Module):
+    def __init__(self, filter_pruner_ctrl: "FilterPruningController", model: nn.Module):  # noqa: F821
```

Comment:
I'm afraid we'll have to insert multiple `noqa`'s into the section where the `pylint` agreed with the code.

Reply:
Only in places where we could not import a corresponding class to have a type hint due to circular dependencies.